pkg/rancher-desktop/config/settings.ts

```diff
@@ -474,114 +473,142 @@ function updateSettings(settings: Settings) {
+
+        locked = JSON.parse(lockedData);
+      } catch {}
+
+      if (typeof defaults !== 'undefined' || typeof locked !== 'undefined') {
+        break;
+      }
     }
     break;
   case 'darwin':
-    try {
-      profileData = fs.readFileSync(join(paths.deploymentProfileSystem, PROFILE_PLIST), 'utf8');
-    } catch {
+    for (const rootPath of [paths.deploymentProfileSystem, paths.deploymentProfileUser]) {
       try {
-        profileData = fs.readFileSync(join(paths.deploymentProfileUser, PROFILE_PLIST), 'utf8');
+        const defaultsData = fs.readFileSync(join(rootPath, 'io.rancherdesktop.profile.defaults.plist'), 'utf8');
+
+        defaults = plist.parse(defaultsData);
       } catch {}
-    }
-    if (typeof profileData !== 'undefined') {
-      profile = plist.parse(profileData);
+      try {
+        const lockedData = fs.readFileSync(join(rootPath, 'io.rancherdesktop.profile.locked.plist'), 'utf8');
+
+        locked = plist.parse(lockedData);
+      } catch {}
+
+      if (typeof defaults !== 'undefined' || typeof locked !== 'undefined') {
+        break;
+      }
     }
     break;
   }
+  defaults = validateDeploymentProfile(defaults, defaultSettings);
+  locked = validateDeploymentProfile(locked, defaultSettings);
 
-  return validateDeploymentProfile(profile, defaultSettings);
+  return { defaults, locked };
 }
 
-function readRegistryDeploymentProfile() {
-  throw new Error('INTERNAL ERROR: readRegistryDeploymentProfile not yet implemented.');
-}
+/**
+ * Windows only. Read settings values from registry using schemaObj as a template.
+ * @param object null - used for recursion.
+ * @param schemaObj the object used as a template for navigating registry.
+ * @param regKey the registry key obtained from nativeReg.openKey().
+ * @param regPath the path to the object relative to regKey.
+ * @returns undefined, or the registry data as an object.
+ */
+function readRegistryUsingSchema(object: any, schemaObj: any, regKey: nativeReg.HKEY, regPath: string[]): any {
+  let regValue;
+  let newObject;
 
-function validateDeploymentProfile(profile: any, schema: any) {
-  if (typeof profile === 'undefined') {
-    return undefined;
-  }
+  for (const [schemaKey, schemaVal] of Object.entries(schemaObj)) {
+    if (typeof schemaVal === 'object' && !Array.isArray(schemaVal)) {
+      regValue = readRegistryUsingSchema(object, schemaVal, regKey, regPath.concat(schemaKey));
+    } else {
+      regValue = nativeReg.getValue(regKey, regPath.join('\\'), schemaKey);
+    }
 
-  Object.keys(profile).forEach((key) => {
-    if (key in schema) {
-      if (typeof profile[key] === typeof schema[key]) {
-        if (typeof profile[key] === 'object') {
-          validateDeploymentProfile(profile[key], schema[key]);
+    if (typeof regValue !== 'undefined' && regValue !== null) {
+      if (typeof newObject === 'undefined') {
+        newObject = Object.create(null);
+      }
+      if (typeof schemaVal === 'boolean') {
```

Comment:
So boolean values are stored as 0/non-0 in the registry?  
If so, I would prefer this code:
```
if (typeof schemaVal === 'boolean') {
  if (typeof regValue === 'number') {
    regValue = regValue !== 0;
  } else {
    console.debug(`Expecting a boolean value for key ${ regPath.concat(schemaKey) }, got "${ regValue }", treating as <false>`);
    regValue = false;
  }
}
```

Reply:
done.